fidesctl/src/fidesctl/core/export.py

```diff
@@ -0,0 +1,219 @@
+"""This module handles the logic required for applying manifest files to the server."""
+from datetime import datetime
+from typing import Dict, List, Tuple, Set
+
+import csv
+
+from fidesctl.core.api_helpers import get_server_resources
+from fidesctl.core.utils import echo_green, get_all_level_fields
+
+
+def list_fides_keys(resources: List) -> List:
+    """
+    Returns a list of fides_keys found in the manifest dir
+    """
+    existing_keys = [resource.fides_key for resource in resources]
+    return existing_keys
+
+
+def export_to_csv(
+    list_to_export: List, resource_exported: str, manifests_dir: str
+) -> str:
+    """
+    Exports a list of Tuples of any length back to the manifest
+    directory as a csv.
+    """
+    utc_datetime = datetime.utcnow().strftime("%Y-%m-%d-T%H%M%S")
+    filename = "".join([utc_datetime, "_", resource_exported, ".csv"])
+    filepath = "/".join([manifests_dir, filename])
+    with open(filepath, "w") as csvfile:
+        writer = csv.writer(csvfile)
+        writer.writerows(list_to_export)
+
+    return filename
+
+
+def generate_data_category_rows(
+    dataset_name: str,
+    dataset_description: str,
+    data_qualifier: str,
+    data_categories: List,
+) -> Set[Tuple[str, str, str, str]]:
+    """
+    Set comprehension to capture categories from any
+    of the possible levels in a dataset resource.
+
+    Returns a set of tuples to be unioned with the
+    overall set to be exported.
+    """
+    dataset_rows = {
+        (
+            dataset_name,
+            dataset_description,
+            category,
+            data_qualifier,
+        )
+        for category in data_categories
+    }
+
+    return dataset_rows
+
+
+def generate_dataset_records(
+    server_dataset_list: List,
+) -> List[Tuple[str, str, str, str]]:
+    """
+    Returns a list of tuples as records (with expected headers)
+    to be exported as csv
+    """
+
+    output_list = [
+        (
+            "dataset.name",
+            "dataset.description",
+            "dataset.data_categories",  # from Kelly - unique data categories per data qualifier, per data use per for the entire dataset, per row.
```

Comment:
can you move this context to the docstring?

Reply:
Fixed! Thank you 🙌🏽